### PR TITLE
Deprecation QubitStateVector

### PR DESCRIPTION
**Pull request by KetpuntoG**
[sc-72046]

### CODE REVIEW

File: tests/ops/functions/test_assert_valid.py [hunk 1]
```diff
@@ -352,6 +356,12 @@ def create_op_instance(c, str_wires=False):
 
 
 @pytest.mark.jax
+@pytest.fixture(scope="function", autouse=True)
+def capture_warnings(recwarn):
+    """Capture warnings."""
+    yield
+
+
```
Comment (albi3ro) [inline tests/ops/functions/test_assert_valid.py:364]: This makes me rather nervous.  I woudl prefer to expliclty catch expected errors with a `with pytest.warns` context manager.

File: tests/ops/qubit/test_attributes.py [hunk 2]
```diff
@@ -151,6 +151,15 @@ class TestSupportsBroadcasting:
...
 
+    @pytest.fixture(scope="function", autouse=False)
+    def capture_warnings(self, recwarn):
+        """Capture warnings."""
+        yield
+        if len(recwarn) > 0:
+            for w in recwarn:
+                assert isinstance(w.message, qml.PennyLaneDeprecationWarning)
+                assert "QubitStateVector is deprecated" in str(w.message)
+
```
Comment (albi3ro) [inline tests/ops/qubit/test_attributes.py:162]: I'd also prefer if we are bit more specific in our warnings control here.

File: tests/ops/functions/test_assert_valid.py (hunk 1 above)
Comment (KetpuntoG) [reply tests/ops/functions/test_assert_valid.py:364]: This test needs to capture warnings from other templates. If I used assert as in other examples, it absorbed them and did not raise the error

File: tests/ops/qubit/test_attributes.py (hunk 2 above)
Comment (KetpuntoG) [reply tests/ops/qubit/test_attributes.py:162]: I'm not sure what you mean here. Do you have any examples of what would be good practice? @albi3ro

Comment (KetpuntoG) [reply tests/ops/qubit/test_attributes.py:162]: @albi3ro

File: tests/ops/functions/test_assert_valid.py [hunk 3]
```diff
@@ -342,7 +342,11 @@ def create_op_instance(c, str_wires=False):
     if dim == 0:
         params = [1] * len(ndim_params)
     elif dim == 1:
-        params = [[1] * 2**n_wires] * len(ndim_params)
+
+        if c == qml.QubitStateVector:
+            params = [[1] + [0] * (2**n_wires - 1)] * len(ndim_params)
+        else:
+            params = [[1] * 2**n_wires] * len(ndim_params)
```
Comment (soranjh) [inline tests/ops/functions/test_assert_valid.py:349]: Why this change is needed in a deprecation PR?

File: tests/ops/functions/test_assert_valid.py [hunk 4]
```diff
@@ -352,6 +356,12 @@ def create_op_instance(c, str_wires=False):
 
 
 @pytest.mark.jax
+@pytest.fixture(scope="function", autouse=True)
```
Comment (soranjh) [inline tests/ops/functions/test_assert_valid.py:359]: Why this change is needed in a deprecation PR?

File: tests/ops/qubit/test_attributes.py [hunk 5]
```diff
@@ -151,6 +151,15 @@ class TestSupportsBroadcasting:
     """Test that all operations in the ``supports_broadcasting`` attribute
     actually support broadcasting."""
 
+    @pytest.fixture(scope="function", autouse=False)
```
Comment (soranjh) [inline tests/ops/qubit/test_attributes.py:154]: Why this change is needed in a deprecation PR?

**Review by soranjh — COMMENTED** [review]
Thanks @KetpuntoG. Some of the changes seem bit irrelevant, left questions on them.

File: tests/ops/functions/test_assert_valid.py (hunk 4 above)
Comment (KetpuntoG) [reply tests/ops/functions/test_assert_valid.py:359]: When there is an operator that you deprecate, it will raise a warning. This is a problem because the tests that use the operator will fail because they have detected a warning that they did not expect before. With this decorator what we are saying is “ignore the deprecation warning”.

I based this on the following [PR](https://github.com/PennyLaneAI/pennylane/pull/6026/files#diff-22bb77302109a21a7718f1b562cc86f1e9061274f76c065f857fcbc4e3202efaR434):

Captura de pantalla 2024-09-09 a las 17 37 57

File: tests/ops/functions/test_assert_valid.py (hunk 3 above)
Comment (KetpuntoG) [reply tests/ops/functions/test_assert_valid.py:349]: The reason is a bit hidden. I had to remove `QubitStateVector` from `tests/ops/functions/conftest.py` because the deprecation warning makes it unable to instantiate.  By removing it from there, the operator will be tested in `def create_op_instance` (this test). This function define some parameters “valid for all templates” but it is not valid for `QubitStateVector`. So I have to manually give a definition of valid parameters

File: tests/ops/functions/test_assert_valid.py (hunk 1 above)
Comment (albi3ro) [reply tests/ops/functions/test_assert_valid.py:364]: Other templates are raising errors about `QubitStateVector`? Can we remove the use of `QubitStateVector` in those other templates?

File: tests/ops/functions/test_assert_valid.py (hunk 3 above)
Comment (albi3ro) [reply tests/ops/functions/test_assert_valid.py:349]: Instead, we can add `QubitStateVector` to either `tests/ops/functions/conftest.py` `_ABSTRACT_OR_META_TYPES` or `_INSTANCES_TO_FAIL`.  Then it will just be skipped during this automatic testing process.

File: tests/ops/qubit/test_attributes.py (hunk 5 above)
Comment (albi3ro) [reply tests/ops/qubit/test_attributes.py:154]: I personally would say we can just skip `QubitStateVector` attribute testing for its deprecation cycle.

File: tests/ops/functions/test_assert_valid.py (hunk 4 above)
Comment (albi3ro) [reply tests/ops/functions/test_assert_valid.py:359]: Preferably, we would just not use `assert_valid` automatic tests on `QubitStateVector` anymore.  We can manually skip it by updating `tests/ops/functions/conftests.py`.

Comment (KetpuntoG) [reply tests/ops/functions/test_assert_valid.py:359]: Oh 😮 I'm going to try that

**Review by AmintorDusko — APPROVED** [review]
Thank you!
